File: backend/app/core/prescription.py
"""Minimum-permission prescription (F19).

Pure function returning the minimal expected permission groups for a category.
Contributes 0 points.
"""

from __future__ import annotations

from app.core.category import expected_groups
from app.core.rules_loader import Rules, get_rules
from app.models.schemas import Prescription


def prescribe_minimum_permissions(
    category_id: str,
    rules: Rules | None = None,
) -> Prescription:
    rules = rules or get_rules()
    expected = expected_groups(category_id, rules)
    sorted_groups = sorted(expected)
    return Prescription(minimal_groups=sorted_groups)

File: shared/src/commonMain/kotlin/com/appautopsy/analysis/core/Prescription.kt
package com.appautopsy.analysis.core

import com.appautopsy.analysis.model.Prescription
import com.appautopsy.analysis.rules.Rules

/**
 * Minimum-permission prescription (F19).
 *
 * The tool does not only diagnose, it prescribes: "a flashlight app can do its
 * job with only: Camera". Derived straight from the category's expected groups,
 * so it can never disagree with the scoring that used the same list.
 *
 * Pure function over the category. Contributes 0 points.
 */
object Prescription {

    fun minimum(categoryId: String, rules: Rules): Prescription =
        Prescription(minimalGroups = Category.expectedGroups(categoryId, rules).sorted())
}
